planner/core/exhaust_physical_plans_test.go

```diff
@@ -201,7 +201,7 @@ func (s *testUnitTestSuit) TestIndexJoinAnalyzeLookUpFilters(c *C) {
 			innerKeys:       []*expression.Column{dsSchema.Columns[1]},
 			pushedDownConds: "a = 1 and c > 'a' and c < 'aaaaaa'",
 			otherConds:      "",
-			ranges:          "[(1 NULL \"a\",1 NULL 0x6161]]",
+			ranges:          "[(1 NULL \"a\",1 NULL \"aa\"]]",
```

Comment:
Why did the range change?

Reply:
it is a bug in the original code. https://github.com/pingcap/tidb/blob/19d3b1f020a4108afbd2351a3da6357ab3e056bb/util/ranger/ranger.go#L517-L520
`SetBytes` will set the Datum type to `KindBytes` which makes the charset to be `binary`. L518 is always false.

Reply:
But the result was changed before this PR dealt with the bug. What is the reason?